packages/lit-html/src/directive-helpers.ts

```diff
@@ -162,9 +162,19 @@ export const insertPart = (
     }
     if (endNode !== refNode || parentChanged) {
       let start: Node | null = part._$startNode;
+      // moveBefore() cannot be called if either the new parent or the node to
+      // move is disconnected, so we fall back to insertBefore() in that case.
```

Comment:
Note that this is not _entirely_ true; as of the latest commits to the spec PR, we throw if exactly one of the old or new parent are disconnected, but we DO support the disconnected->disconnected case, in response to some developer feedback. Just FYI!